test/envtest/konnect_entities_kongconsumer_test.go

```diff
@@ -1058,4 +1059,396 @@ func TestKongConsumerSecretCredentials(t *testing.T) {
+				return req.HMACAuthID == hmacID && req.ConsumerIDForNestedEntities == consumerID
+			}),
+		).Return(nil, nil)
+
+		t.Log("Creating a KongCredentialHMACAuth to adopt the existing HMAC auth")
+		createdHMACAuth := &configurationv1alpha1.KongCredentialHMAC{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "hmac",
+			},
+			Spec: configurationv1alpha1.KongCredentialHMACSpec{
+				Adopt: &commonv1alpha1.AdoptOptions{
+					From: commonv1alpha1.AdoptSourceKonnect,
+					Mode: commonv1alpha1.AdoptModeOverride,
+					Konnect: &commonv1alpha1.AdoptKonnectOptions{
+						ID: hmacID,
+					},
+				},
+				ConsumerRef: corev1.LocalObjectReference{
+					Name: createdConsumer.Name,
+				},
+				KongCredentialHMACAPISpec: configurationv1alpha1.KongCredentialHMACAPISpec{
+					Username: lo.ToPtr("user"),
+					Secret:   lo.ToPtr("secret"),
+				},
+			},
+		}
+		require.NoError(t, clientNamespaced.Create(ctx, createdHMACAuth))
+
+		t.Log("Waiting for the KongCredentialHMAC to be programmed and set Konnect ID")
+		watchFor(t, ctx, wHMAC, apiwatch.Modified, func(hmac *configurationv1alpha1.KongCredentialHMAC) bool {
+			return hmac.Name == createdHMACAuth.Name && k8sutils.IsProgrammed(hmac) && hmac.GetKonnectID() == hmacID
+		},
+			"KongCredentialHMAC did not get programmed or set Konnect ID",
+		)
+	})
+
+	// Adopting a JWT
+	t.Run("JWT", func(t *testing.T) {
+		jwtID := uuid.NewString()
+		wJWT := setupWatch[configurationv1alpha1.KongCredentialJWTList](t, ctx, cl, client.InNamespace(ns.Name))
+
+		t.Log("Setting up SDK expectations for getting and updating JWTAuths")
+		sdk.KongCredentialsJWTSDK.EXPECT().GetJwtWithConsumer(
+			mock.Anything,
+			mock.MatchedBy(func(req sdkkonnectops.GetJwtWithConsumerRequest) bool {
+				return req.JWTID == jwtID && req.ConsumerIDForNestedEntities == consumerID
+			}),
+		).Return(&sdkkonnectops.GetJwtWithConsumerResponse{
+			Jwt: &sdkkonnectcomp.Jwt{
+				ID: lo.ToPtr(jwtID),
+				Consumer: &sdkkonnectcomp.JWTConsumer{
+					ID: lo.ToPtr(consumerID),
+				},
+				Algorithm: lo.ToPtr(sdkkonnectcomp.AlgorithmHs256),
+				Key:       lo.ToPtr("jwt-key"),
+				Secret:    lo.ToPtr("jwt-secret"),
+			},
+		}, nil)
+		sdk.KongCredentialsJWTSDK.EXPECT().UpsertJwtWithConsumer(
+			mock.Anything,
+			mock.MatchedBy(func(req sdkkonnectops.UpsertJwtWithConsumerRequest) bool {
+				return req.JWTID == jwtID && req.ConsumerIDForNestedEntities == consumerID
+			}),
+		).Return(nil, nil)
+
+		t.Log("Creating a KongCredentialJWT for adopting the existing JWT auth")
+		createdJWT := &configurationv1alpha1.KongCredentialJWT{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "jwt",
+			},
+			Spec: configurationv1alpha1.KongCredentialJWTSpec{
+				Adopt: &commonv1alpha1.AdoptOptions{
+					From: commonv1alpha1.AdoptSourceKonnect,
+					Mode: commonv1alpha1.AdoptModeOverride,
+					Konnect: &commonv1alpha1.AdoptKonnectOptions{
+						ID: jwtID,
+					},
+				},
```

Comment:
This is OK to be used as is in this PR but I think we can extract this to a generate function in a separate PR and refactor all the usages of `AdoptOptions` to use that new function. WDYT?

Reply:
Yes. I think we could move the deployment of the credentials into dedicated functions.